docs/docs/dev_docs/contracts/syntax/historical_trees.md

```diff
@@ -0,0 +1,227 @@
+---
+title: Historical Access Tree
+---
+
+The Aztec Protocol uses an append-only Merkle tree to store the headers of all previous blocks in the chain as its leaves. You can learn more about how it works in the [concepts section](../../../concepts/advanced/data_structures/trees.md#historical-access-tree).
+
+On this page you will learn how you can integrate inclusion and non-inclusion proofs into your own smart contract.
+
+# Historical tree library
+
+The historical tree library allows you to access:
+
+* [Private notes](#note-inclusion)
+* [Notes that have been nullified](#nullifier-inclusion)
+* [Notes that have not been nullified](#note-validity)
+* [Public values](#public-value-inclusion)
+* [Contracts](#contract-inclusion)
+
+Using the historical access tree, you can check that specific notes or nullifiers happened at specific blocks. This can be useful for things such as:
+
+* Verifying a timestamp that was created in a private context
+* Checking eligibility based on historical events (eg for an airdrop) 
+
+# How to integrate historical access and inclusion proofs into your smart contract
+
+Checking if a value was included in a certain block is called an inclusion proof. Similarly, checking that a value was not included is called a non-inclusion proof.
+
+In this section we will talk about how to integrate the different types of inclusion and non-inclusion proofs into your smart contract.
+
+## Note inclusion 
+
+Note inclusion proves that someone owned a note at a specific block number.
+
+### 1. Import note_inclusion into your smart contract
+
+```rust
+aztec::{
+history::{
+    #include_code import_note_inclusion yarn-project/noir-contracts/contracts/inclusion_proofs_contract/src/main.nr raw
+}
+}
+```
+### 2. Call prove_note_inclusion
+
+`prove_note_inclusion` takes 4 parameters:
+
+| Name            | Type                   | Description                                         |
+|-----------------|------------------------|-----------------------------------------------------|
+| note_interface  | NoteInterface<Note, N> | Interface for the note with necessary functionality|
+| note_with_header| Note                   | The note you are proving inclusion for             |
+| block_number    | u32                    | Block number for proving note's existence           |
+| context         | PrivateContext         | Private context     |
+
+#### Example
+
+#include_code prove_note_inclusion yarn-project/noir-contracts/contracts/inclusion_proofs_contract/src/main.nr rust
+
+### 3. You can also prove the note commitment with prove_note_commitment_inclusion
```

Comment:
What is the difference between proving a note and a note commitment? I understand notes to be commitments, but clearly my understanding is not differentiated enough.